stdlib/list.mli

```diff
@@ -102,6 +102,39 @@ val flatten : 'a list list -> 'a list
 (** An alias for [concat]. *)
 
 
+(** {1 Comparison} *)
+
+val equal : ('a -> 'a -> bool) -> 'a list -> 'a list -> bool
```

Comment:
What about:
````ocaml
val equal : ('a -> 'b -> bool) -> 'a list -> 'b list -> bool
````
This might be a bit weird for an equality function, but it also makes it more useful (and it makes it clear that it's just a variant of `for_all2`).